### PR TITLE
Decouple workflows controller from Fedora 3

### LINKED ISSUE #2400: Retrieve pid and version of objects from dor-services-app in workflows_controller
Remove the retrieval of the objects from Fedora 3 in load_resource and retrieve pid and version from dor-services-app

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Fixes #2400

### CODE REVIEW

File: app/components/workflow_process_row.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 class WorkflowProcessRow < ApplicationComponent
   # @param [Dor::Workflow::Response::Process] process_status the model for the WorkflowProcess
   # @param [Integer] index the row index
-  # @param [Dor::Item] item the repository object that the workflow is about
+  # @param [Cocina::Models::DRO] item the repository object that the workflow is about
   def initialize(process:, index:, item:)
```
Comment (mjgiarlo) [inline app/components/workflow_process_row.rb:8]: I know it changes a public interface, but should `item` here change to `cocina_item` per our new cocina-models usage guidance?

File: app/controllers/workflows_controller.rb [hunk 2]
```diff
@@ -59,6 +57,8 @@ def update
 
   # add a workflow to an object if the workflow is not present in the active table
   def create
+    cocina = maybe_load_cocina(params[:item_id])
```
Comment (mjgiarlo) [inline app/controllers/workflows_controller.rb:60]: Will this always be an item? Or can it be an APO or collection? I ask mostly to help guide what variable name we use here.

File: app/presenters/workflow_presenter.rb [hunk 3]
```diff
@@ -3,9 +3,11 @@
 class WorkflowPresenter
   # @param [Object] view the rails view context
   # @param [WorkflowStatus] workflow_status
-  def initialize(view:, workflow_status:)
+  # @param [Cocina::Models::DRO] item the repository object that the workflow is about
+  def initialize(view:, workflow_status:, item:)
```
Comment (mjgiarlo) [inline app/presenters/workflow_presenter.rb:7]: Should `item` be `cocina_item` or `cocina_object` instead?

**Review by mjgiarlo — COMMENTED** [review]
Looks good overall, but have a question about variable naming with regard to cocina model instances.

File: app/controllers/workflows_controller.rb (hunk 2 above)
Comment (jcoyne) [reply app/controllers/workflows_controller.rb:60]: It could be a Collection or a DRO, both have workflows.

Comment (mjgiarlo) [reply app/controllers/workflows_controller.rb:60]: OK. Should we call it `cocina_object` then?

**Review by jmartin-sul — COMMENTED** [review]
agreed with @mjgiarlo, looks good to me overall, but might be nice to switch from `item` to `cocina_object` across the board

**Comment by jcoyne** [comment]
This already uses cocina_object. Am I missing something?

File: app/presenters/workflow_presenter.rb (hunk 3 above)
Comment (mjgiarlo) [reply app/presenters/workflow_presenter.rb:7]: @jcoyne here's the one that's outstanding

Comment (jcoyne) [reply app/presenters/workflow_presenter.rb]: Thank you. I've got it.

**Review by jmartin-sul — APPROVED** [review]
thanks for the ivar/param renaming!

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**